### PR TITLE
Fix BackButton handling escape before all other elements

### LINKED ISSUE #5177: SearchTextBox no longer clears text when pressing Escape
**Describe the bug:** The expected behaviour, when pressing `Esc`, is that the `SearchTextBox` on the carousel clears the text it contains. After that, the next `Esc` key pressed should return to the Menu.

**Screenshots or videos showing encountered issue:** https://streamable.com/116sk

**osu!lazer version:** 2019.628.0

### PR DESCRIPTION
**Pull request by nyquillerium**
fixes https://github.com/ppy/osu/issues/5177

It seems that other than `OsuTextBox` and `FocusedTextBox`, only overlays use `GlobalAction.Back` when there's the possibility of a back button present. 

I think its safe to simply move the back button to the end of the queue, where it will allow any potential UI elements to handle it first.

### CODE REVIEW

**Comment by peppy** [comment]
Is there not a more local way to do this? We should be avoiding using linq like this (every frame) as much as possible.

**Comment by nyquillerium** [comment]
One way I can think of is overriding `HandleNonPositionalInput` (since that queue gets used for building the `KeybindingInputQueue`, but that is internal.

Also, I believe this particular keybinding container doesn't create this queue every frame, as putting a break point on it only triggers the linq expression when I actually press escape. FrameworkActionContainer does trigger every frame in my testing.

File: osu.Game/Input/Bindings/GlobalActionContainer.cs [hunk 1]
```diff
@@ -55,8 +56,11 @@ public GlobalActionContainer(OsuGameBase game)
             new KeyBinding(new[] { InputKey.Control, InputKey.Minus }, GlobalAction.DecreaseScrollSpeed),
         };
 
+        /// <summary>
+        /// Make sure that the <see cref="OsuGame"/> handles global input first, and that <see cref="BackButton"/> handles global input last.
+        /// </summary>
         protected override IEnumerable<Drawable> KeyBindingInputQueue =>
-            handler == null ? base.KeyBindingInputQueue : base.KeyBindingInputQueue.Prepend(handler);
+            (handler == null ? base.KeyBindingInputQueue : base.KeyBindingInputQueue.Prepend(handler)).OrderBy(d => d is BackButton);
```
Comment (smoogipoo) [inline osu.Game/Input/Bindings/GlobalActionContainer.cs]: Instead of this, how about putting `BackButton` at the very deepest level, and proxying it to the above layer?

```
Children =
    backButton
    /* ScreenStack */
    backButton.CreateProxy()
```

**Review by smoogipoo — CHANGES_REQUESTED** [review]

**Comment by peppy** [comment]
Are you aware you moved the button to a different scaling context in this change?

**Comment by nyquillerium** [comment]
I noticed that when I made the change, but in my testing scaling remained the same, so I assumed it was alright.

**Comment by nyquillerium** [comment]
nevermind that, this needs further work

File: osu.Game/Screens/Select/Footer.cs [hunk 2]
```diff
@@ -104,8 +104,8 @@ public Footer()
             updateModeLight();
         }
 
-        protected override bool OnMouseDown(MouseDownEvent e) => true;
+        protected override bool OnMouseDown(MouseDownEvent e) => false;
```
Comment (peppy) [inline osu.Game/Screens/Select/Footer.cs]: why are these here

Comment (nyquillerium) [reply osu.Game/Screens/Select/Footer.cs]: I'm pretty sure those don't exist on the current diff?

File: osu.Game/OsuGame.cs [hunk 3]
```diff
@@ -417,7 +416,16 @@ protected override void LoadComplete()
                                     screenStack.Exit();
                             }
                         },
-                        logoContainer = new Container { RelativeSizeAxes = Axes.Both },
+                        new Container
```
Comment (peppy) [inline osu.Game/OsuGame.cs]: Is there a particular reason to add this container level?

Comment (nyquillerium) [reply osu.Game/OsuGame.cs]: This is resolving the original issue by moving the back button one level higher than the screen stack, while allowing the back button to remain contained inside the `screenContainer`. Not sure if there's another way to achieve this.

Comment (peppy) [reply osu.Game/OsuGame.cs]: It doesn't work with just ordering?

Comment (nyquillerium) [reply osu.Game/OsuGame.cs]: didn't think of that, seems to work fine.

**Comment by Joehuu** [comment]
`BackButton` doesn't do its hover animation (i.e. it is blocked) when `ModSelectOverlay` is open.

**Review by peppy — APPROVED** [review]

**Comment by peppy** [comment]
@Joehuu please open that as a separate issue. Will need some further restructuring I believe.
